journal/src/main/java/io/camunda/zeebe/journal/file/SegmentsManager.java

```diff
@@ -295,6 +304,10 @@ private void prepareNextSegment() {
     nextSegment = CompletableFuture.supplyAsync(() -> createUninitializedSegment(descriptor));
   }
 
+  Collection<Segment> getSegments(final long index) {
+    return Collections.unmodifiableSortedMap(segments.tailMap(index, true)).values();
```

Comment:
I wasn't 100% sure about the `inclusive = true` here, but I think it's safer to do that.